### PR TITLE
Add useAuthSignOutAction hook

### PR DESCRIPTION
**Pull request by reesscot**
Adds an internal hook to trigger an Auth Sign Out action. 

*Example use*:
```jsx

export const SignOutButton = () => {

  const authSignOutAction = useAuthSignOutAction({ global: true });

  return (
    <Button onClick={authSignOutAction}>
      Sign out
    </Button>
  );
};
```

*Example of reloading the page in response to action completion:*
```js
Hub.listen("ui-actions", (capsule) => {
  if (capsule.payload.event === "AuthSignOut_Finished") {
    window.location.reload();
  }
});
```

*Todo*

- [x] Add unit tests

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/react/src/hooks/actions/useAuthSignOutAction.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+interface UseAuthSignOutAction {
+  (opts?: SignOutOpts): () => Promise<void>;
+}
+
+export const useAuthSignOutAction: UseAuthSignOutAction =
+  (opts) => async () => {
+    try {
+      Hub.dispatch(HUB_AUTH_SIGNOUT_NAMESPACE, {
+        event: ACTION_EVENT_STARTED,
```
Comment (alharris-at) [inline packages/react/src/hooks/actions/useAuthSignOutAction.tsx:19]: Don't you want the opts also attached to this as well? We should probably indicate whether or not it was global in the callback hooks.

**Review by alharris-at — COMMENTED** [review]
Overall LGTM, probably want to add unit tests as well?

File: packages/react/src/hooks/actions/useAuthSignOutAction.tsx (hunk 1 above)
Comment (reesscot) [reply packages/react/src/hooks/actions/useAuthSignOutAction.tsx:19]: Good idea, will update

**Review by alharris-at — APPROVED** [review]

File: packages/react/src/hooks/actions/constants.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const ACTIONS_CHANNEL = 'ui-actions';
+export const ACTION_AUTH_SIGNOUT = 'AuthSignOut';
+
+export const ACTION_STARTED = 'Started';
+export const ACTION_FINISHED = 'Finished';
+
+export const ACTION_AUTH_SIGNOUT_STARTED = `${ACTION_AUTH_SIGNOUT}.${ACTION_STARTED}`;
```
Comment (zchenwei) [inline packages/react/src/hooks/actions/constants.ts:7]: nit: The real string `AuthSignOut.Started` may look a bit weird. It seems `AuthSignOut_Started` has a good looking IMO

File: packages/react/src/internal.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 export * from './hooks/useAuth';
 export * from './hooks/useDataStore';
 export * from './hooks/useStorageURL';
+export * from './hooks/actions/useAuthSignOutAction';
```
Comment (zchenwei) [inline packages/react/src/internal.tsx:4]: Should we start using explicit export?

File: packages/react/src/hooks/actions/constants.ts (hunk 2 above)
Comment (reesscot) [reply packages/react/src/hooks/actions/constants.ts:7]: Yeah, that makes sense to me. Will update

File: packages/react/src/internal.tsx (hunk 3 above)
Comment (reesscot) [reply packages/react/src/internal.tsx:4]: Yeah, that's probably safer. Will update

File: packages/react/src/hooks/actions/__tests__/useAuthSignOutAction.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    await authSignOutAction();
+    expect(signOutSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call Hub with started and finished events', async () => {
+    const authSignOutAction = useAuthSignOutAction();
+
+    await authSignOutAction();
+    expect(hubDispatchSpy).toHaveBeenCalledTimes(2);
+    expect(hubDispatchSpy).toHaveBeenCalledWith('ui-actions', {
```
Comment (zchenwei) [inline packages/react/src/hooks/actions/__tests__/useAuthSignOutAction.test.ts:29]: nit: `ui-action` could be replaced with `ACTIONS_CHANNEL`

**Review by zchenwei — APPROVED** [review]
LGTM! Just one non-blocking comment

**Pull request merged**